src/components/ArtifactTimeline.vue

```diff
@@ -119,7 +112,8 @@
       expanded.value = [val]
     }
     getArtifacts()
-  })
+  }, { immediate: true })
```

Comment:
You're calling getArtifacts in the onBeforeMount and then again here in this watcher. Adding immediate means it's going to get artifacts twice. Is that intentional?

Reply:
Ah actually that's an oversight; we should only refetch artifacts if this _isn't_ the first iteration. Will update, nice catch